### PR TITLE
Expose all memory stats contained in memory.stat via API and /metrics

### LINKED ISSUE #975: Export RSS size in /metrics
The metric exported as container_memory_usage_bytes includes page cache. It would be nice if cadvisor also exported the RSS size, which is often more interesting.

### PR DESCRIPTION
**Pull request by guoshimin**
fixes #975

### CODE REVIEW

File: metrics/prometheus.go [hunk 1]
```diff
@@ -127,6 +127,21 @@ func NewPrometheusCollector(infoProvider infoProvider, f ContainerNameToLabelsFu
 					return values
 				},
 			}, {
+				name:        "container_memory_stats",
+				help:        `Various memory statistics. See the section "5.2 stat file" of https://www.kernel.org/doc/Documentation/cgroups/memory.txt for details`,
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{"stat_name"},
```
Comment (jimmidyson) [inline metrics/prometheus.go]: nit: can we just make this `name`?

**Comment by jimmidyson** [comment]
One nit, but otherwise LGTM.

**Comment by jimmidyson** [comment]
ok to test

**Comment by jimmidyson** [comment]
@pwittrock How do I trigger Jenkins builds now?

File: info/v1/container.go [hunk 2]
```diff
@@ -306,6 +306,11 @@ type MemoryStats struct {
 	// Units: Bytes.
 	Usage uint64 `json:"usage"`
 
+	// Various statistics reported in memory.stat in cgroupfs. See the section
```
Comment (vishh) [inline info/v1/container.go]: We already include some of the stats from `memory.stat`. For example `WorkingSet` is derived using data from `memory.stat`.
Can we instead expose specific fields? Explicit fields will help with understanding of the API as well.

File: info/v1/container.go [hunk 3]
```diff
@@ -306,6 +306,11 @@ type MemoryStats struct {
...
+	// "5.2 stat file" of https://www.kernel.org/doc/Documentation/cgroups/memory.txt
+	// for details.
+	Stats map[string]uint64 `json:"stats"`
```
Comment (vishh) [inline info/v1/container.go]: We will have to update `v2` API as well.

File: info/v1/container.go (hunk 2 above)
Comment (guoshimin) [inline info/v1/container.go]: All this is just plumbing to get to the information in memory.stat, which is officially documented at https://www.kernel.org/doc/Documentation/cgroups/memory.txt. I don't find it tasteful to manually enumerate the entries here and duplicate the official documentation.

As a compromise, I can limit the scope of the PR to only expose RSS, which is the original request in #975.

File: info/v1/container.go (hunk 3 above)
Comment (guoshimin) [inline info/v1/container.go]: The v2 API references v1: https://github.com/google/cadvisor/blob/e4262b91b17cc132a35a023865f9ef97a048aa21/info/v2/container.go#L102

File: info/v1/container.go (hunk 2 above)
Comment (vishh) [inline info/v1/container.go]: Yeah. Let's limit to RSS for now. We get quite a few questions/issues around the data exposed by cAdvisor. Honestly, all the data that cAdvisor exposes is documented as part of kernel APIs.

File: info/v1/container.go (hunk 3 above)
Comment (vishh) [inline info/v1/container.go]: My bad. :+1:

File: info/v1/container.go (hunk 2 above)
Comment (dqminh) [inline info/v1/container.go]: I would like to also see at least `cache` exposed.

**Comment by guoshimin** [comment]
Opened another PR #1055. Closing this one.

**Pull request closed without merging**